backend/packages/Upgrade/src/api/services/FeatureFlagService.ts

```diff
@@ -405,34 +452,71 @@ export class FeatureFlagService {
-        where: { segment: { id: segmentId } },
-        relations: ['featureFlag', 'segment'],
-      });
-    }
+    await this.createDeleteListAuditLogs([segmentId], filterType, currentUser);
+    return this.segmentService.deleteSegment(segmentId, logger);
+  }
 
-    // delete list AuditLogs here
-    const updateAuditLog: FeatureFlagUpdatedData = {
-      flagId: existingRecord.featureFlag.id,
-      flagName: existingRecord.featureFlag.name,
-      list: {
-        listId: segmentId,
-        listName: existingRecord.segment.name,
-        filterType: filterType,
-        operation: FEATURE_FLAG_LIST_OPERATION.DELETED,
-      },
-    };
+  async createDeleteListAuditLogs(
+    segmentIds: string[],
+    filterType: FEATURE_FLAG_LIST_FILTER_MODE,
+    currentUser: User,
+    entityManager?: EntityManager
+  ): Promise<void> {
+    const executeTransaction = async (manager: EntityManager) => {
+      const auditLogPromises = [];
 
-    await this.experimentAuditLogRepository.saveRawJson(LOG_TYPE.FEATURE_FLAG_UPDATED, updateAuditLog, currentUser);
+      for (const segmentId of segmentIds) {
+        let existingRecord: FeatureFlagSegmentInclusion | FeatureFlagSegmentExclusion;
 
-    return this.segmentService.deleteSegment(segmentId, logger);
+        if (filterType === FEATURE_FLAG_LIST_FILTER_MODE.INCLUSION) {
+          existingRecord = await manager.findOne(FeatureFlagSegmentInclusion, {
+            where: { segment: { id: segmentId } },
+            relations: ['featureFlag', 'segment'],
+          });
+        } else {
+          existingRecord = await manager.findOne(FeatureFlagSegmentExclusion, {
+            where: { segment: { id: segmentId } },
+            relations: ['featureFlag', 'segment'],
+          });
+        }
+
+        // Handle if the record is not found
+        if (!existingRecord) {
+          throw new Error(`Segment with ID ${segmentId} not found for ${filterType}`);
+        }
+
+        // Create the delete list audit log data
+        const updateAuditLog: FeatureFlagUpdatedData = {
+          flagId: existingRecord.featureFlag.id,
+          flagName: existingRecord.featureFlag.name,
+          list: {
+            listId: segmentId,
+            listName: existingRecord.segment.name,
+            filterType: filterType,
+            operation: FEATURE_FLAG_LIST_OPERATION.DELETED,
+          },
+        };
+
+        const savePromise = manager.save(ExperimentAuditLog, {
+          type: LOG_TYPE.FEATURE_FLAG_UPDATED,
+          data: updateAuditLog,
+          user: currentUser,
+        });
+
+        auditLogPromises.push(savePromise);
+      }
+
+      // Use Promise.all to run all audit log saving operations concurrently
+      await Promise.all(auditLogPromises);
+    };
+
+    if (entityManager) {
+      await executeTransaction(entityManager);
+    } else {
+      await this.dataSource.transaction(async (manager) => {
```

Comment:
You should not start a new transaction. Use the repository.

Reply:
code updated